text/0249-v2-experiments.md

```diff
@@ -408,3 +484,240 @@ Disadvantages:
+| pip   | yes         | yes                              | no                 | yes                         |
+| NuGet | ?           | ?                                | ?                  | ?                           |
+| Go    | ?           | ?                                | ?                  | ?                           |
+
+**NPM**
+
+Stable ranges: yes, `^1.60.0` does not include `1.61.1-experimental`.
+
+Experimental app uses stable lib: can be done, the lib only uses a peerDependency. `npm ls` will complain about
+a missing peerDependency though.
+
+**Maven**
+
+Stable ranges: no, `[1.60.0,)` *will* include `1.61.1-experimental`.
+
+Experimental app uses stable lib:
+
+**pip**
+
+Stable ranges: yes, `>=1.60.0` will not match version `1.60.1-experimental` [ref](https://pip.pypa.io/en/stable/reference/pip_install/#pre-release-versions)
+
+Experimental app uses stable lib: `pip` allows overriding transitive dependencies using `requirements.txt`, but it will complain
+while doing so:
+
+Example (I tried with different packages):
+
+```text
+awscli 1.18.158 has requirement s3transfer<0.4.0,>=0.3.0, but you'll have s3transfer 0.2.0 which is incompatible.
+```
+
+**NuGet**
+
+...pending responses from .NET SDK team...
+
+## Different major versions
+
+We would vend the experimental build under a prerelease tag, for example:
+
+- `aws-cdk-lib@2.60.0`
+- `aws-cdk-lib@3.60.0` (odd ranges are experimental)
+
+Or:
+
+- `aws-cdk-lib@2.60.0`
+- `aws-cdk-lib@102.60.0` (100+ versions are experimental)
+
+| .     | Stable apps | Experimental app uses Stable lib | Without complaints | Lib advertises Experimental |
+|-------|-------------|----------------------------------|--------------------|-----------------------------|
+| NPM   | yes         | yes                              | yes                | no                          |
+| Maven | yes         | with excluded dependencies       | yes                | yes                         |
+| pip   | yes         | yes                              | no                 | yes                         |
+| NuGet | ?           | ?                                | ?                  | ?                           |
+| Go    | ?           | ?                                | ?                  | ?                           |
+
+**NPM**
+
+Mostly like pre-release tags, except libraries using stable can explicitly declare they're usable
+with both stable and experimental ranges by using multiple `peerDependency` ranges:
+
+```json
+{
+  "peerDependencies": {
+    "aws-cdk-lib": "^2.60.0,^102.60.0"
+  }
+}
+```
+
+This would suppress the warning you would otherwise get for a non-compatible version.
+
+**Maven**
+
+Like pre-release tags, except we don't get the mixing in of pre-release versions with
+regular versions.
+
+**pip**
+
+Like pre-release tags.
+
+**NuGet**
+
+...pending responses from .NET SDK team...
```

Comment:
AFAIK, .NET does not support supplying multiple version constraints... All my attempts so far resulted in constraints being entirely ignored, resulting in .NET resolving to the lowest existing version (not the desired behavior).